internal/kafka/internal/services/clusters.go

```diff
@@ -70,6 +71,7 @@ type ClusterService interface {
 	Delete(cluster *api.Cluster) (bool, *apiErrors.ServiceError)
 	ConfigureAndSaveIdentityProvider(cluster *api.Cluster, identityProviderInfo types.IdentityProviderInfo) (*api.Cluster, *apiErrors.ServiceError)
 	ApplyResources(cluster *api.Cluster, resources types.ResourceSet) *apiErrors.ServiceError
+	RemoveResources(cluster *api.Cluster, syncSetName string) *apiErrors.ServiceError
```

Comment:
As a side comment, I believe we should start returning `error` instead of `*ServiceError` to avoid the `nil is not nil` error.

Reply:
I see this more as a design issue than a specific issue with this method. Which has to do with whether we want the consumers of the service to have accessible details of the ServiceError or perform the conversion by themselves. Whatever the decision is all the methods in the interface should be consistent.

Reply:
returning a custom object exposes us to the `nil is not nil` error.
```go

type mycustomerror struct{}

func (mycustomerror) Error() string {
	return ""
}

func myfunc1() *mycustomerror {
	return nil
}

func myfunc2() error {
	return nil
}

func myfunc3() error {
	err := myfunc2()
	err = myfunc1()
	return err
}

func main() {
	if myfunc3() == nil {
                 // Never enters here
		fmt.Println("No error")
	} else {
                // always error even if err value is `nil`!!
		fmt.Println("Error!", err)
	}
}
```

On the other hand if we always return `error` and we always return `naked nils` the issue goes away:
```
type mycustomerror struct{}

func (mycustomerror) Error() string {
	return ""
}

func (mycustomerror) Error() string {
	return ""
}

func myfunc1() error {
	return nil
}

func myfunc2() error {
	return nil
}

func myfunc3() error {
	err := myfunc2()
	err = myfunc1()
	return err
}

func main() {
	err := myfunc3()
	if myfunc3() == nil {
		fmt.Println("No error")
	} else {
		fmt.Println("Error!", err)
	}
}
```

Since returning an `error` is safer, I believe we should move to that approach.

> Whatever the decision is all the methods in the interface should be consistent.
+1